packages/open-next/src/adapters/routing.ts

```diff
@@ -0,0 +1,104 @@
+import { InternalEvent } from './event-mapper.js';
+import { debug } from './logger.js';
+import {
+  RedirectDefinition,
+  RewriteDefinition,
+  RewriteMatcher,
+  RoutesManifest,
+} from './next-types.js';
+
+const redirectMatcher =
+  (
+    headers: Record<string, string>,
+    cookies: Record<string, string>,
+    query: Record<string, string | string[]>
+  ) =>
+  (redirect: RewriteMatcher) => {
+    switch (redirect.type) {
+      case 'header':
+        console.log(
+          'redirect-header',
+          headers?.[redirect.key.toLowerCase()],
+          redirect.value
+        );
+        return (
+          headers?.[redirect.key.toLowerCase()] &&
+          new RegExp(redirect.value ?? '').test(
+            headers[redirect.key.toLowerCase()] ?? ''
+          )
+        );
+      case 'cookie':
+        return (
+          cookies?.[redirect.key] &&
+          new RegExp(redirect.value ?? '').test(cookies[redirect.key] ?? '')
+        );
+      case 'query':
+        return query[redirect.key] && Array.isArray(redirect.value)
+          ? redirect.value.reduce(
```

Comment:
I'm confused by this reduce, `true` is initially passed in, so `true || x` will always be true?

Reply:
You're right, just a silly mistake